spec/fixtures/modules/upload_file/files/subdir/sub-file.sh

```diff
@@ -0,0 +1 @@
+Sub-content!
```

Comment:
Another approach would be to embed some heredocs to create whatever filesystem objects you might need?

Reply:
That seems about equivalent to using fixtures here, to me. If anything, on-disk fixtures are easier to find/debug.